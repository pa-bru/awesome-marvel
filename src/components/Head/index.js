import React from 'react'
import PropTypes from 'prop-types'
import Helmet from 'react-helmet'

export default function Head({ title, description, image, children }) {
  return (
    <Helmet>
      <title>{title}</title>
      <meta name="description" content={description} />
      <meta name="og:title" content={title} />
      <meta name="og:description" content={description} />
      <meta name="twitter:title" content={title} />
      <meta name="twitter:description" content={description} />
      <meta name="twitter:image" content={image ? image : `${process.env.PUBLIC_URL}/icon.png`} />
      <meta name="og:image" content={image ? image : `${process.env.PUBLIC_URL}/icon.png`} />
      {children}
    </Helmet>
  )
}

Head.propTypes = {
  title: PropTypes.string,
  description: PropTypes.string,
  image: PropTypes.string,
  children: PropTypes.node,
}
