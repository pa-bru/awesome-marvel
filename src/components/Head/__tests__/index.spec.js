import React from 'react'
import { shallow } from 'enzyme'
import Head from '../'

describe('Head', () => {
  it('should render correctly', () => {
    // When
    const tree = shallow(<Head />)

    // Then
    expect(tree).toMatchSnapshot()
  })

  it('should render with a child', () => {
    // When
    const tree = shallow(
      <Head>
        <meta property="og:title" content="additional meta tag for the Head component" />
      </Head>
    )

    // Then
    expect(tree).toMatchSnapshot()
  })
})
