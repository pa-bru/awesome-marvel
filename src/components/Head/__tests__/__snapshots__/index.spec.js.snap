// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Head should render correctly 1`] = `
<HelmetWrapper
  defer={true}
  encodeSpecialCharacters={true}
>
  <title />
  <meta
    name="description"
  />
  <meta
    name="og:title"
  />
  <meta
    name="og:description"
  />
  <meta
    name="twitter:title"
  />
  <meta
    name="twitter:description"
  />
  <meta
    content="/icon.png"
    name="twitter:image"
  />
  <meta
    content="/icon.png"
    name="og:image"
  />
</HelmetWrapper>
`;

exports[`Head should render with a child 1`] = `
<HelmetWrapper
  defer={true}
  encodeSpecialCharacters={true}
>
  <title />
  <meta
    name="description"
  />
  <meta
    name="og:title"
  />
  <meta
    name="og:description"
  />
  <meta
    name="twitter:title"
  />
  <meta
    name="twitter:description"
  />
  <meta
    content="/icon.png"
    name="twitter:image"
  />
  <meta
    content="/icon.png"
    name="og:image"
  />
  <meta
    content="additional meta tag for the Head component"
    property="og:title"
  />
</HelmetWrapper>
`;
